USAGE.md

```diff
@@ -80,6 +80,28 @@ Packwerk reads from the `packwerk.yml` configuration file in the root directory.
 | cache                | false                                     | when true, caches the results of parsing files |
 | cache_directory      | tmp/cache/packwerk                        | the directory that will hold the packwerk cache |
 
+### Using custom parsers
+
+You can specify a custom parser to parse different file formats (e.g. slim or haml)
+
+```ruby
+class SlimParser
+  include ParserInterface
```

Comment:
Yes, that makes sense 👍  
the Ruby parser also defines the `Parser` namespace, to avoid confusion, I added `Packerk::` to every usage of this interface.

Reply:
I wonder if we should rename to something like `FileParser`?